mplc/multi_partner_learning.py

```diff
@@ -139,13 +140,14 @@ def log_partner_perf(self, partner_id, partner_index, history):
 
         logger.debug(f"{epoch_nb_str} > {mb_nb_str} > {partner_id_str} > val_acc: {val_acc_str}")
 
-    def eval_and_log_model_val_perf(self):
+    def evaluate_model(self, x, y):
         model = self.build_model()
-        hist = model.evaluate(self.val_data[0],
-                              self.val_data[1],
-                              batch_size=constants.DEFAULT_BATCH_SIZE,
-                              verbose=0,
-                              )
+        hist = model.evaluate(x, y, batch_size=constants.DEFAULT_BATCH_SIZE, verbose=0)
+
+        return hist
+
```

Comment:
I disagree with this changes. For most of the mpl method, it is just a copy of the keras' one. 
Furthermore, you are not evaluating a model, but a mpl, which I found a little bit strange. 
Finally, it allows you to create a strange mpl method, which do not return any usable model.

The model saved by your method is a nn, which hasn't been fitten and which is far away from the objective of your method. Same for the return of the build_model() method. 

The change of the .evaluate() should be done at the model level I think
For instance you could create a class EnsembleModel

```python
class AveragedModel( ? ):
   def __init__(list_of_partner_model): 
         self.list_of_partner_model = list_of_partner_model
  
   def fit(x, y)
         pass #or maybe train all the partners' models with the new data 
  
   def predict(x):
     predictions_list = []
        for partner in self.partners_list:

            model = partner.build_model()
            predictions = model.predict(x)
            predictions_list.append(predictions)

          return np.mean(predictions_list, axis=0)  # by the way, by curiosity can it be done with keras or sklearn ? 
def evaluate(x, y):

# and so on , with the save function, etc
```

Reply:
maybe with a [AvgPooling](https://keras.io/api/layers/pooling_layers/average_pooling1d/) layer, or [ensemble method](https://scikit-learn.org/stable/modules/ensemble.html)

It is not really important